xdsl/transforms/experimental/ConvertStencilToLLMLIR.py

```diff
@@ -358,10 +362,61 @@ def map_returns(op: Operation) -> None:
     return return_targets
 
 
+_OpT = TypeVar('_OpT', bound=Operation)
+
+
+def all_matching_uses(op_res: Iterable[SSAValue],
+                      typ: type[_OpT]) -> Iterable[_OpT]:
+    for res in op_res:
+        for use in res.uses:
+            if isinstance(use.operation, typ):
+                yield use.operation
+
+
+def infer_halo_from_load_op(op: LoadOp) -> tuple[IndexAttr, IndexAttr]:
+    applies: list[ApplyOp] = list(all_matching_uses([op.res], ApplyOp))
+    assert len(applies) > 0, "Load must be followed by Apply!"
+
+    shape_lb: None | IndexAttr = None
+    shape_ub: None | IndexAttr = None
+
+    for apply in applies:
+        stores: list[StoreOp] = list(all_matching_uses(apply.res, StoreOp))
+
+        assert len(stores) > 0, "Apply must be followed by store!"
+        for store in stores:
+            lb, ub = store.lb, store.ub
+            if shape_lb is None:
+                shape_lb = lb
+                shape_ub = ub
+                continue
+
+            shape_lb = IndexAttr.min(lb, shape_lb)
+            shape_ub = IndexAttr.max(ub, shape_ub)
```

Comment:
Because this happens a lot, min and max do this themselves 🙂

Reply:
Yep, noticed! Thanks!